pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -244,13 +245,19 @@ private void startReceivingMessages(List<ConsumerImpl<T>> newConsumers) {
     }
 
     private void receiveMessageFromConsumer(ConsumerImpl<T> consumer) {
-        consumer.receiveAsync().thenAcceptAsync(message -> {
+        CompletableFuture<List<Message<T>>> messagesFuture;
+        if (consumer.numMessagesInQueue() >= 10) {
```

Comment:
Why 10?

Reply:
Just make sure It's worth using batch receive API. If less than 10 messages are in the internal consumer and the batch receive operation will also create an ArrayList. And I think we'd better not introduce a new configuration to make the client configuration more complex.